payments/models.py

```diff
@@ -135,6 +135,71 @@ def get_success_url(self) -> str:
     def get_process_url(self) -> str:
         return reverse('process_payment', kwargs={'token': self.token})
 
+    def get_payment_url(self):
+        """
+        Get the url the view that handles the payment (payment_details() in documentation)
+        For now used only by PayU provider to redirect users back to CVV2 form
+        """
+        raise NotImplementedError()
+
+    def get_user(self):
+        """ Get the user asociated with this payment """
+        raise NotImplementedError()
+
+    def get_user_email(self):
```

Comment:
What's wrong with `billing_email`? What's expected of this method?

Reply:
The `billing_*` fields hold information about the issuer (Me), PayU (and potentially other providers) needs information about the user.

Reply:
The `billing_` fields all hold billing details, that is: the details of the person paying (this is what payment providers usually ask for, and the thing that actually varies on a per-payment basis).

This is the assumption made for other payment implementations, so it's the assumption that any third-party providers should make two.

Using `billing_` fields as something elsemeans that applications cannot use PayU **and** another provider, since the give different semantic meaning to the same fields.

Reply:
Oh, this is something, I got totally wrong. Sorry for loss of your time and thank you for clearing that out.
I used this for storing issuer info when I was implementing PayPal (which doesn't use it at all), and then continued with the wrong implementation for PayU and recurring payments.

Maybe it would be useful to put some info about those fields into docs, I will try to write something.